src/online2bin/online2-tcp-nnet3-decode-faster.cc

```diff
@@ -265,7 +294,17 @@ int main(int argc, char *argv[]) {
             if (decoder.NumFramesDecoded() > 0) {
               Lattice lat;
               decoder.GetBestPath(false, &lat);
+              TopSort(&lat); // for LatticeStateTimes(),
               std::string msg = LatticeToString(lat, *word_syms);
+
+              // get time-span after previous endpoint,
+              if (produce_time) {
+                int32 t_beg = frame_offset;
+                int32 t_end = frame_offset + GetLatticeTimeSpan(lat);
```

Comment:
I'm not sure if `TopSort` and `LatticeStateTimes` are really needed here. Can't we compute `t_end` using `decoder.NumFramesDecoded()` and/or `samp_count`/`check_count`?

Reply:
yes, these are good points, the PR is changed now accordingly...